License namespace model

tjasmith opened the pull request:
I have added the following:
- license namespace model
- organisation model
- updated the navbar to accomodate a new menu
- Added the submit license namespace form, and template

techytushar commented on src/src/secret.py:
```diff
@@ -1,8 +1,8 @@
 def getGithubKey():
-    return 'GHKEYXXX'
+    return '376d679b554ae9a9d474'
```
Be careful and don't commit your GitHub App ID and Secret Key, review your changes before pushing them

tjasmith replied:
Thanks for pointing that out. I have corrected this.

techytushar commented on src/src/secret.py:
```diff
@@ -1,8 +1,8 @@
 
 def getGithubSecret():
-    return 'GHSECRETXXX'
+    return 'b0b172b5780173783a4c86d60bae983c4608d6a7'
```
Same here!!

tjasmith replied:
Thanks for pointing that out. I have corrected this.

techytushar commented:
There is still work left to be done, leave a comment when you think its ready for review, nice work so far :+1:

tjasmith commented:
@techytushar Thanks,
I pushed this so you all can see my progress and suggest changes which I can implement as I progress.

goneall commented:
Looks good so far - I like the dropdown for the license requests - solves some horizontal real estate issues.

tjasmith commented:
@techytushar @goneall 

- I have added a method to verify whether a license is already present on the spdx license list
- added the possibility to create an organisation(via the admin ui)
- added namespace id to the model and form
- added the publiclyShared boolean to the form

goneall commented on src/app/utils.py:
```diff
@@ -203,3 +205,54 @@ def check_license_name(name):
+    "exists": False
+    }
+    for license in license_list["licenses"]:
+        if namespaceId == license["licenseId"] or namespace == license["name"]:
```
You can also compare the license text to see if it matches the listed licenses.  See https://github.com/spdx/spdx-online-tools/blob/ecd4426e8ccdd5ecd79fae10ede31142f900f77a/src/app/views.py#L880 for the code to do this.

Note that  this method is currently rather slow.  I'm planning on making some changes to speed this up over the next month or so.

techytushar commented:
@tjasmith you branch has conflicts with the `master` branch, please update your branch with master, resolve the conflicts and push again.

goneall commented:
Looking good so far - @tjasmith @techytushar Let me know if you want me to merge any code into master

techytushar commented on src/app/utils.py:
```diff
@@ -203,3 +205,53 @@ def check_license_name(name):
+    license_list = getLicenseList(token)
+    return_dict = {
+    "name": "",
+    "licenseId": "",
```
There is no need to define all the keys in the `return_dict` beforehand, instead you can only add the `exists` keys to the dict and the rest will be added from lines `243 . . .` if the license matches.

tjasmith replied:
@techytushar You want me to make the URL field required?

techytushar replied:
No, to apply a check on the URL field, so that user enters a valid URL, as it happens in the e-mail field.

tjasmith replied:
Ok, I have done that.

techytushar commented on src/app/views.py:
```diff
@@ -202,10 +310,23 @@ def createIssue(licenseAuthorName, licenseName, licenseIdentifier, licenseCommen
+    return r.status_code
+
+
+def createLicenseNamespaceIssue(licenseNamespace, token, urlType):
```
This function can be defined in the `utils` file and then imported in the `views` file

tjasmith commented:
@techytushar I have corrected the code as you mentionned.

tjasmith commented:
@techytushar @goneall Please could you have a look at this?

goneall commented:
I noticed that there is a merge conflict with src/app/review - @tjasmith if you could resolve the merge conflict, I'll go ahead and merge.

tjasmith commented:
@goneall Thanks for pointing that out. I have resolved the merge conflict.
